multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/validators/parameters/RoutePartValidator.java

```diff
@@ -1,22 +1,37 @@
 package org.cloudfoundry.multiapps.controller.core.validators.parameters;
 
+import java.util.Collections;
 import java.util.Locale;
 import java.util.Map;
+import java.util.Set;
 
 import org.cloudfoundry.multiapps.common.SLException;
+import org.cloudfoundry.multiapps.common.util.MapUtil;
+import org.cloudfoundry.multiapps.controller.core.model.SupportedParameters;
 import org.cloudfoundry.multiapps.controller.core.util.NameUtil;
 import org.cloudfoundry.multiapps.mta.model.Module;
 
 public abstract class RoutePartValidator implements ParameterValidator {
 
+    protected final String namespace;
+    protected final boolean applyNamespaceGlobal;
+
+    public RoutePartValidator(String namespace, boolean applyNamespaceGlobal) {
+        this.namespace = namespace;
+        this.applyNamespaceGlobal = applyNamespaceGlobal;
+    }
+
     @Override
     public String attemptToCorrect(Object routePart, final Map<String, Object> context) {
         if (!(routePart instanceof String)) {
             throw new SLException(getErrorMessage(), routePart);
         }
 
-        String result = (String) routePart;
-        result = NameUtil.getNameWithProperLength(result, getRoutePartMaxLength());
+        // FIXME This logic does not work in case where apply-namespace is set to false
```

Comment:
same here

Reply:
Actually this is fixed with next commit "Add namespace to routes parameters only"

Reply:
yeah, forgot to delete this comment as I saw the other commit afterwards